services/fluxmonitor.go

```diff
@@ -0,0 +1,272 @@
+package services
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"sort"
+	"time"
+
+	"github.com/shopspring/decimal"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/tidwall/gjson"
+)
+
+type FluxMonitorConfig struct {
+	Adapters          []url.URL
+	From              string
+	To                string
+	Multiply          int64
+	Threshold         decimal.Decimal
+	AbsoluteThreshold decimal.Decimal
+	Heartbeat         time.Duration
+	PollInterval      time.Duration
+}
+
+func ParseFMSpec(jsonSpec json.RawMessage) FluxMonitorConfig {
+	var fmConfig FluxMonitorConfig
+	res := gjson.GetBytes(jsonSpec, "feeds.#.url")
+	var adapters []url.URL
+	for _, adapter := range res.Array() {
+		url, _ := url.Parse(adapter.String())
+		adapters = append(adapters, *url)
+	}
+	fmConfig.Adapters = adapters
+	fmConfig.From = gjson.GetBytes(jsonSpec, "requestData.data.from").String()
+	fmConfig.To = gjson.GetBytes(jsonSpec, "requestData.data.to").String()
+	fmConfig.Multiply = gjson.GetBytes(jsonSpec, "precision").Int()
+	fmConfig.Threshold, _ = decimal.NewFromString(gjson.GetBytes(jsonSpec, "threshold").String())
+	fmConfig.AbsoluteThreshold, _ = decimal.NewFromString(gjson.GetBytes(jsonSpec, "absoluteThreshold").String())
+	fmConfig.Heartbeat, _ = time.ParseDuration(gjson.GetBytes(jsonSpec, "idleTimer.duration").String())
+	fmConfig.PollInterval, _ = time.ParseDuration(gjson.GetBytes(jsonSpec, "pollTimer.period").String())
+	return fmConfig
+}
+
+type FluxAggregatorState struct {
+	currentAnswer decimal.Decimal
+}
+
+type AdapterResponse struct {
+	Price decimal.Decimal `json:"result"`
+	// might need error response as well
+}
+type FluxMonitor struct {
+	state  FluxAggregatorState
+	config FluxMonitorConfig
+
+	// subscriber subscriber.ISubscriber
+	latestResult decimal.Decimal
+
+	// quitOnce     sync.Once
+
+	// chBlockchainEvents chan subscriber.Event
+	chDeviation chan decimal.Decimal
+	// chNewround  chan FluxAggregatorState
+	// chClose chan struct{}
+}
+
+func NewFluxMonitor(config FluxMonitorConfig, triggerJobRun chan decimal.Decimal) *FluxMonitor {
```

Comment:
Instead of taking a `chan decimal.Decimal` here, it should be of type `subscriber.Event`, where we get the blockchain manager to take the FM state data and convert it to a job run. This is because each blockchain's external adapter might take different input params. (e.g. Substrate will need the feed_id, while EVM chains will need an address)